vpr/src/place/centroid_move_generator.cpp

```diff
@@ -35,5 +36,19 @@ e_create_move CentroidMoveGenerator::propose_move(t_pl_blocks_to_be_moved& block
         return e_create_move::ABORT;
     }
 
-    return ::create_move(blocks_affected, b_from, to);
+    e_create_move create_move;
+    create_move = ::create_move(blocks_affected, b_from, to);
+
+    //Check if the move is legal from a floorplan perspective
+    //Check that all of the blocks affected by the move would still be in a legal floorplan region after the swap
+    bool floorplan_legal = true;
```

Comment:
make this chunk of lines a helper function in place_constraints.cpp so that you can just get this with one line

Reply:
and pass blocks_affected by const reference

Reply:
can make it an inline function in the header file - say it is performance critical